### ISSUE TITLE
[BUG/ISSUE] Cannot save out a subset of the DryDepMix or DryDepChm diagnostics

### ISSUE THREAD

**Issue by msulprizio** [geoschem/geos-chem#722]
In both GCClassic and GCHP, when I try to save out the DryDepMix and/or DryDepChm diagnostics for one species, the model crashes with an out-of-bounds error. For example, adding the following lines to the DryDep collection:

```
'DryDepMix_O3            ', 'GCHPchem',
'DryDepChm_O3            ', 'GCHPchem',
```

returns:

```
At line 160 of file CodeDir/src/GCHP_GridComp/GEOSChem_GridComp/geos-chem/GeosCore/diagnostics_mod.F90
Fortran runtime error: Index '2' of dimension 3 of array 'state_diag%drydepchm' above upper bound of 1
```

In GEOS-Chem Classic, if I change the lines in HISTORY.rc to:

```
'DryDepMix_?DRY?            ', 'GIGCchem',
'DryDepChm_?DRY?           ', 'GIGCchem',
```

the run finishes and saves out the diagnostics successfully. Adding lines for each drydep species in GCHP also allows the run to finish and save out the diagnostics successfully.

I used GEOS-Chem 13.1.0-alpha.6 and gfortran 9.3 in my test runs.

**Comment by yantosca** [comment]
@msulprizio: The error at line 160 is in the `State_Diag%FracOfTimeInTrop`.  But this should be a 3-D array.

**Comment by lizziel** [comment]
@yantosca, I think you might looking at the wrong version. In the dev branch line 16 is this: https://github.com/geoschem/geos-chem/blob/440e458735924be078c12447c916d5792871255a/GeosCore/diagnostics_mod.F90#L160

**Comment by msulprizio** [comment]
@lizziel Thanks, that does make more sense. I think the dynamic array allocation is wrong for DryDepMix and DryDepChm. I requested one species, so it's allocating one space in the third dimension but uses species ID, in this case O3=2, causing an OOB error.

**Comment by lizziel** [comment]
Yeah, I think DryDep, DryDepChm, and DryDepMix should all use the same mapping, and it should be the union of all species found for the all of them. Looks like instead there is separate mapping for each, a problem when DryDep needs to sum Chm and Mix.

**Comment by yantosca** [comment]
@lizziel, oops, you're right.  Sorry about that.
